rpc/server.go

```diff
@@ -51,25 +61,51 @@ type Server struct {
 	addr           net.Addr              // Server address; may change if picking unused port
 	closed         bool                  // Set upon invocation of Close()
 	closeCallbacks []func(conn net.Conn) // Slice of callbacks to invoke on conn close
+	methods        map[string]method
 }
 
 // NewServer creates a new instance of Server.
 func NewServer(addr net.Addr, context *Context) *Server {
 	s := &Server{
-		Server:  rpc.NewServer(),
 		context: context,
 		addr:    addr,
+		methods: map[string]method{},
 	}
 	heartbeat := &HeartbeatService{
 		clock:              context.localClock,
 		remoteClockMonitor: context.RemoteClocks,
 	}
-	if err := s.RegisterName("Heartbeat", heartbeat); err != nil {
+	if err := heartbeat.Register(s); err != nil {
 		log.Fatalf("unable to register heartbeat service with RPC server: %s", err)
 	}
 	return s
 }
 
+// Register a new method handler. `name` is a qualified name of the form "Service.Name".
+// `handler` is a function that takes an argument of the same type as `reqPrototype`.
+// Both the argument and return value of 'handler' should be a pointer to a protocol
```

Comment:
That verification happens down in the codec itself (although verifying it here will have much cleaner error handling).